svo (curing skeleton, controllers, action system).xml

```diff
@@ -2237,34 +2237,130 @@ function svo.sendcuring(what)
+        end
       end
-      what = "curing priority"..table.concat(sk.affqueue)
-      sk.sendqueuecmd(what)
-      sk.affqueue = {}
     end
-  elseif type == 'defence' then
-    if not table.contains(sk.defqueue, prio) then
-      table.insert(sk.defqueue, prio)
+    
+    if sk.sendpriotimer then killTimer( sk.sendpriotimer ) end
+  
+    if #sk.affqueue &gt; 0 then
+        local what = "curing priority"..table.concat(sk.affqueue)
+        if sk.queuecount &gt; 4 then
+          tempTimer(1,function()
+            sendstuff(what,"normal")
+            sk.affqueue = {}
+            sk.queuecount = 0
+          end)
+        else
+          sendstuff(what,"normal")
+          sk.affqueue = {}
+          sk.queuecount = sk.queuecount + 1
+        end
     end
-    if #sk.defqueue &gt;= 20 then
-      if #sk.sendqueue &gt;= 6 then
-        sk.dosendqueue()
-      end
-      what = "curing priority defence"..table.concat(sk.defqueue)
-      sk.sendqueuecmd(what)
-      sk.defqueue = {}
+    if #sk.defqueue &gt; 0 then
+        local what = "curing priority defence"..table.concat(sk.defqueue)
+        if sk.queuecount &gt; 4 then
+          tempTimer(1,function()
+          sendstuff(what,"normal")
+          sk.defqueue = {}
+          sk.queuecount = 0
+          end)
+        else
+          sendstuff(what,"normal")
+          sk.defqueue = {}
+          sk.queuecount = sk.queuecount + 1
+        end
+    end
+    if #sk.affqueueslow &gt; 0 then
+        local what = "curing priority"..table.concat(sk.affqueueslow)
+        if sk.queuecount &gt; 4 then
+          tempTimer(1,function()
+          sendstuff(what,"slow")
+          sk.affqueueslow = {}
+          sk.queuecount = 0
+          end)
+        else
+          sendstuff(what,"slow")
+          sk.affqueueslow = {}
+          sk.queuecount = sk.queuecount + 1
+        end
+    end
+    if #sk.defqueueslow &gt; 0 then
+        local what = "curing priority defence"..table.concat(sk.defqueueslow)
+        if sk.queuecount &gt; 4 then
+          tempTimer(1,function()
+          sendstuff(what,"slow")
+          sk.defqueueslow = {}
+          sk.queuecount = 0
+          end)
+        else
+          sendstuff(what,"slow")
+          sk.defqueueslow = {}
+          sk.queuecount = sk.queuecount + 1
+          display(sk.queuecount)
+        end
     end
+  
+  if queuetimer then killTimer(queuetimer) end
+  local queuetimer = tempTimer(1,function() sk.queuecount = 0 end)     
+  sk.sendpriotimer = nil
+end
+
+function svo.sendcuringprio(prio, type, slow)
+  local sys = svo.sys
+  prio = " "..prio
+  local what, tab, cmd
+  if type == 'affliction' then
+    tab = "affqueue"
+    cmd = "curing priority"
+  elseif type == 'defence' then
+    tab = "defqueue"
+    cmd = "curing priority defence"
   else
+    return
+  end
+  if slow == true then tab = tab.."slow" end
+  if ( table_length( sk[tab] ) + #cmd + #prio ) &gt; sk.achaea_command_max_length then
```

Comment:
This is highly inefficient (and lazy), replace it with a table that keeps the length of the currently queued command to send. See `sk.sendqueuel` for an example of this.